src/sdam/state/mod.rs

```diff
@@ -477,15 +506,23 @@ impl TopologyState {
             topology.clone(),
             self.http_client.clone(),
         );
+        let is_load_balanced = matches!(topology.client_options().load_balanced, Some(true));
+        if is_load_balanced {
+            // Because load balancer servers don't have a monitoring connection, the associated
+            // connection pool needs to be directly marked as ready.
+            let server = Arc::clone(&server);
+            RUNTIME.execute(async move {
```

Comment:
Since this task is running in the background, there could be a potential race between the first operation and this mark as ready completing. I think we could avoid this if we the pool just marked itself as ready in `new` depending on if we're in load balanced mode or not.

Reply:
Done; also realized that `Monitor` was already storing the `ClientOptions`, so `Monitor::start` didn't need the `is_load_balanced` parameter.